test/mock-agent.js

```diff
@@ -142,89 +142,6 @@ describe('MockAgent - dispatch', () => {
       onError: () => {}
     }))
   })
-
-  test('should throw if handler is not valid on redirect', (t) => {
-    t = tspl(t, { plan: 7 })
-
-    const baseUrl = 'http://localhost:9999'
-
-    const mockAgent = new MockAgent()
-    after(() => mockAgent.close())
-
-    t.throws(() => mockAgent.dispatch({
-      origin: baseUrl,
-      path: '/foo',
-      method: 'GET'
-    }, {
-      onError: 'INVALID'
-    }), new InvalidArgumentError('invalid onError method'))
-
-    t.throws(() => mockAgent.dispatch({
-      origin: baseUrl,
-      path: '/foo',
-      method: 'GET'
-    }, {
-      onError: (err) => { throw err },
-      onConnect: 'INVALID'
-    }), new InvalidArgumentError('invalid onConnect method'))
-
-    t.throws(() => mockAgent.dispatch({
-      origin: baseUrl,
-      path: '/foo',
-      method: 'GET'
-    }, {
-      onError: (err) => { throw err },
-      onConnect: () => {},
-      onBodySent: 'INVALID'
-    }), new InvalidArgumentError('invalid onBodySent method'))
-
-    t.throws(() => mockAgent.dispatch({
-      origin: baseUrl,
-      path: '/foo',
-      method: 'CONNECT'
-    }, {
-      onError: (err) => { throw err },
-      onConnect: () => {},
-      onBodySent: () => {},
-      onUpgrade: 'INVALID'
-    }), new InvalidArgumentError('invalid onUpgrade method'))
-
-    t.throws(() => mockAgent.dispatch({
-      origin: baseUrl,
-      path: '/foo',
-      method: 'GET'
-    }, {
-      onError: (err) => { throw err },
-      onConnect: () => {},
-      onBodySent: () => {},
-      onHeaders: 'INVALID'
-    }), new InvalidArgumentError('invalid onHeaders method'))
-
-    t.throws(() => mockAgent.dispatch({
-      origin: baseUrl,
-      path: '/foo',
-      method: 'GET'
-    }, {
-      onError: (err) => { throw err },
-      onConnect: () => {},
-      onBodySent: () => {},
-      onHeaders: () => {},
-      onData: 'INVALID'
-    }), new InvalidArgumentError('invalid onData method'))
-
-    t.throws(() => mockAgent.dispatch({
-      origin: baseUrl,
-      path: '/foo',
-      method: 'GET'
-    }, {
-      onError: (err) => { throw err },
-      onConnect: () => {},
-      onBodySent: () => {},
-      onHeaders: () => {},
-      onData: () => {},
-      onComplete: 'INVALID'
-    }), new InvalidArgumentError('invalid onComplete method'))
-  })
```

Comment:
why was this removed?

Reply:
The redirect interceptor has some kind of assertion of the handler that does not exist anywhere else and is difficult to have compat with through the wrap/unwrap.